### PR TITLE
specifications: Architecture overview clarification

### LINKED ISSUE #80: [Qualcomm feedback] Chapter 6. Architectural Overview - IOMMUs security domain specific
Reference: https://lists.riscv.org/g/tech-ap-tee-io/topic/103498833#47

> CoVE-IO uses
the Smmtt I/O MTT extension and the platform IOMMUs security domain specific Register
Programming Interfaces (RPI) 

what is the terminology in IOMMU spec? should it say supervisor domain specific?  where can we find details of IOMMU RPI?

### LINKED ISSUE #81: [Qualcomm feedback] Chapter 6. Architectural Overview - TSM to configure
Reference: https://lists.riscv.org/g/tech-ap-tee-io/topic/103498833#47

> CoVE-IO relies on the TSM to configure both the PCIe physical device and the root port
it’s attached to in order to establish IDE streams between the two. The PCIe root port IDE
configuration from the TSM is abstracted through the platform Root-of-Trust (RoT)

This paragraph is not clear as to who configures what. Could you clarify. What is platform RoT referring to?

### PR DESCRIPTION
**Pull request by sameo**
Fixes #80
Fixes #81

### CODE REVIEW

File: specification/06-arch_overview.adoc [hunk 1]
```diff
@@ -43,7 +44,9 @@ The device attestation process is specific to each TVM workload configuration,
 but it may be handled by the Trusted Device Manager (TDM).
 
 In order to protect the confidentiality and integrity of TVM data between the
-PCIe root port and a bound TDI, CoVE-IO relies on the TSM to configure both the
-PCIe physical device and the root port it's attached to in order to establish
-IDE streams between the two. The PCIe root port IDE configuration from the TSM
-is abstracted through the platform Root-of-Trust (RoT).
+PCIe root port and a bound TDI, PCIe IDE streams are established between them.
+CoVE-IO relies on the TSM to configure the IDE keys and capabilities for both
```
Comment (jyao1) [inline specification/06-arch_overview.adoc:48]: > TSM to configure the IDE keys and capabilities

I think TSM can only configure the IDE keys, but not it cannot configure the capabilities.

File: specification/06-arch_overview.adoc [hunk 2]
```diff
@@ -27,12 +27,13 @@ and registering IOMMUs and PCIe root ports, and then binding physical devices
...
 With the CoVE-IO ABIs and flows, TDIs can access TVM confidential memory
-directly. Based on the TVM configuration, the confidential DMA memory
-could be all TVM memory or a subset of the TVM memory.
-CoVE-IO uses the Smmtt I/O MTT extension and the platform IOMMUs
-security domain specific Register Programming Interfaces (RPI) to grant TDIs
-with direct access to their bound TVM confidential memory and isolate it from
-DMA originating from any unbound TDI.
+directly through DMA. Based on the TVM configuration, the DMA accessible
+confidential memory could be all TVM memory or a subset of the TVM memory.
+A platform IOMMU instance must be tied to security domains holding TVMs to
```
Comment (jyao1) [inline specification/06-arch_overview.adoc:32]: A platform may have multiple IOMMU instances.

File: specification/06-arch_overview.adoc (hunk 1 above)
Comment (sameo) [reply specification/06-arch_overview.adoc]: Fixed.

File: specification/06-arch_overview.adoc (hunk 2 above)
Comment (sameo) [reply specification/06-arch_overview.adoc]: Yes. I meant that each TSM domain should hold one of those instances. I tried to clarify that statement now.

File: specification/06-arch_overview.adoc [hunk 3]
```diff
@@ -43,7 +44,9 @@ The device attestation process is specific to each TVM workload configuration,
...
-PCIe root port and a bound TDI, CoVE-IO relies on the TSM to configure both the
-PCIe physical device and the root port it's attached to in order to establish
-IDE streams between the two. The PCIe root port IDE configuration from the TSM
-is abstracted through the platform Root-of-Trust (RoT).
+PCIe root port and a bound TDI, PCIe IDE streams are established between them.
+CoVE-IO relies on the TSM to configure the IDE keys and capabilities for both
+the PCIe physical device and the PCIe root port the device is attached to. To
+configure the PCIe root port IDE keys while keeping the TSM implementation
+platform agnostic, the TSM sends IDE Key Management (KM) data objects to the
+platform Root-of-Trust (RoT).
```
[the remark this answers is not in the document]
Comment (sameo) [reply specification/06-arch_overview.adoc]: I fixed that now.

File: specification/06-arch_overview.adoc [hunk 4]
```diff
@@ -27,12 +27,14 @@ and registering IOMMUs and PCIe root ports, and then binding physical devices
...
 With the CoVE-IO ABIs and flows, TDIs can access TVM confidential memory
-directly. Based on the TVM configuration, the confidential DMA memory
-could be all TVM memory or a subset of the TVM memory.
-CoVE-IO uses the Smmtt I/O MTT extension and the platform IOMMUs
-security domain specific Register Programming Interfaces (RPI) to grant TDIs
-with direct access to their bound TVM confidential memory and isolate it from
-DMA originating from any unbound TDI.
+directly through DMA. Based on the TVM configuration, the DMA accessible
+confidential memory could be all TVM memory or a subset of the TVM memory. At
+least one platform IOMMU instance must be exclusively tied to each security
```
Comment (rsahita) [inline specification/06-arch_overview.adoc:32]: Security domain -> supervisor domain

File: specification/06-arch_overview.adoc [hunk 5]
```diff
@@ -43,7 +45,12 @@ The device attestation process is specific to each TVM workload configuration,
...
-PCIe physical device and the root port it's attached to in order to establish
-IDE streams between the two. The PCIe root port IDE configuration from the TSM
-is abstracted through the platform Root-of-Trust (RoT).
+PCIe root port and a bound TDI, PCIe IDE streams are established between them.
+CoVE-IO relies on the TSM to generate and configure the IDE keys for both the
+PCIe physical device and the PCIe root port the device is attached to. For
+setting the device IDE keys up, the TSM must use the IDE Key Management
+(`IDE_KM`) protocol. For the PCIe root port, the TSM must protect the IDE keys
+exchange through e.g. a SPDM session with the platform Root-of-Trust (RoT) DOE
+mailbox, when it supports that mechanism. If the communication between TSM and
```
Comment (rsahita) [inline specification/06-arch_overview.adoc:54]: For this you added the variant where the TSM programs the PCie root directly- does that PR supersede this one?

**Review by rsahita — CHANGES_REQUESTED** [review]
Lgtm with minor fixes

File: specification/06-arch_overview.adoc (hunk 4 above)
Comment (sameo) [reply specification/06-arch_overview.adoc]: Thanks. Fixed now.

**Comment by sameo** [comment]
> Lgtm with minor fixes

@rsahita Let me know if the changes look good to you.

File: specification/06-arch_overview.adoc (hunk 5 above)
Comment (rsahita) [reply specification/06-arch_overview.adoc:54]: sg

**Review by rsahita — APPROVED** [review]

**Pull request merged**